File: js/preview-content/DetectEnvironment.js
// DocSection: preview_content_detect_environment
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
const KontentDelivery = require('@kentico/kontent-delivery');

// TODO: Determine whether the app is running in a preview environment
const isPreview = ...;

const deliveryClient = new KontentDelivery.DeliveryClient({
    projectId: '<YOUR_PROJECT_ID',
    previewApiKey: '<YOUR_PREVIEW_API_KEY>',
    globalQueryConfig:  {
        usePreviewMode: isPreview, // Queries the Delivery Preview API.
    },
    typeResolvers: []
});
// EndDocSection

File: java/preview-content/DetectEnvironment.java
// DocSection: preview_content_detect_environment
// Tip: Find more about Java/JavaRx SDKs at https://docs.kontent.ai/java
import com.github.kentico.kontent.delivery;

// TODO: Determine whether the app is running in a preview environment
boolean isPreview = ...;

DeliveryOptions deliveryOptions = new DeliveryOptions();
deliveryOptions.setProjectId("<YOUR_PROJECT_ID>");
deliveryOptions.setPreviewApiKey("<YOUR_PREVIEW_API_KEY>");
deliveryOptions.setUsePreviewApi(isPreview ? true : false);

DeliveryClient client = new DeliveryClient(deliveryOptions);
// EndDocSection